packages/flutter_tools/lib/src/commands/android_analyze.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../convert.dart';
+import '../project.dart';
+
+enum AndroidAnalyzeOption {
+  listBuildVariant,
+  outputAppLinkSettings,
+}
+
+class AndroidAnalyze {
+  AndroidAnalyze({
+    required this.fileSystem,
+    required this.option,
+    required this.userPath,
+    this.buildVariant,
+    required this.logger,
+  });
+
+  final FileSystem fileSystem;
+  final AndroidAnalyzeOption option;
+  final String? buildVariant;
+  final String userPath;
+  final Logger logger;
+
+  Future<void> analyze() async {
+    final FlutterProject project = FlutterProject.fromDirectory(fileSystem.directory(userPath));
+    await _analyze(project);
+  }
+
+  Future<void> _analyze(FlutterProject project) async {
+    final String result;
+    switch (option) {
+      case AndroidAnalyzeOption.listBuildVariant:
+        result = jsonEncode(await project.android.getBuildVariants());
+        logger.printStatus('output = $result');
```

Comment:
this will be machine-parsed, right? If so, why do you need the `output = ` bit?

Reply:
I need some identifier as this calling this command may create some unwanted noise, for example, `downloading flutter tool...` or `running gradle task....`.

Or do you have a better way for other tool that uses this command to get the output? An alternative is do the same thing as the outputAppLinkDomains to dump a file, but I am not sure if that will be overkill...

Reply:
I **think** what clients of the `flutter daemon` protocol do is ignore any line that cannot be parsed as JSON. @DanTup do I have that right?

Reply:
also, do we know that `jsonEncode` will always output a single line, even for arbitrarily long lists?

Reply:
> I think what clients of the flutter daemon protocol do is ignore any line that cannot be parsed as JSON. @DanTup do I have that right?

Yep. It's a bit dirty really.. ideally it would be an entirely JSON interface, but whenever you call a `flutter` command, there's always the possibility that the shell script will print stuff like that it's downloading components, so we always have to handle non-JSON and drop it (or in the case of `flutter run` we print it to the Debug Console).

Reply:
remove the output keyword

> also, do we know that jsonEncode will always output a single line, even for arbitrarily long lists?

I don't think they will do that unless you pretty print. Even if they do, it can still be parsed as json

Reply:
> remove the output keyword
> 
> > also, do we know that jsonEncode will always output a single line, even for arbitrarily long lists?
> 
> I don't think they will do that unless you pretty print. Even if they do, it can still be parsed as json

but if it's on multiple lines, knowing how to select just the JSON part would be non-trivial, right?

Reply:
true. but it is doable I think.

I just check the source code, the only possibility that the encode contains newline is when they use JsonEncoder.withIndent, which is not used by the jsonEncode.

Reply:
SGTM